### PR TITLE
Post external smoke test results to GitHub PR

### LINKED ISSUE #30615: Output results from running e2e tests on external site to a comment on a PR
Once work on https://github.com/woocommerce/woocommerce/issues/30614 is completed, we want to look at how we can output the results into a comment on the PR.

### PR DESCRIPTION
**Pull request by MrJnrman**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR takes the results from running smoke tests against an external site and posts it as a comment on the associated PR. This action get triggered when the `run: smoke tests` label is added to the PR.

Closes #30615 .

### How to test the changes in this Pull Request:

1. Create a PR
2. Add `run: smoke tests` label
3. Wait for tests to complete and observe comments on the associated PR

### CODE REVIEW

File: .github/workflows/pr-smoke-test.yml [hunk 1]
```diff
@@ -49,8 +51,50 @@ jobs:
           DEFAULT_TIMEOUT_OVERRIDE: 120000
         run: |
           pnpx wc-e2e test:e2e plugins/woocommerce/tests/e2e/specs/smoke-tests/update-woocommerce.js
+
+      - name: Post Smoke tests results comment on PR
+        if: always()
+        uses: actions/github-script@v5
+        env: 
+          TITLE: 'External Smoke Test Resutls:'
```
Comment (rrennick) [inline .github/workflows/pr-smoke-test.yml]:
```suggestion
          TITLE: 'Smoke Test Results:'
```

File: .github/workflows/pr-smoke-test.yml [hunk 2]
```diff
@@ -49,8 +51,50 @@ jobs:
...
+          UPDATE_WC: 1
+          DEFAULT_TIMEOUT_OVERRIDE: 120000
+        run: |
           pnpx wc-e2e test:e2e
 
+      - name: Post E2E tests results comment on PR
+        if: always()
+        uses: actions/github-script@v5
+        env: 
+          TITLE: 'External E2E Test Resutls:'
```
Comment (rrennick) [inline .github/workflows/pr-smoke-test.yml]: Same suggested change here

File: .github/workflows/pr-smoke-test.yml [hunk 3]
```diff
@@ -49,8 +51,50 @@ jobs:
...
+      - name: Post Smoke tests results comment on PR
+        if: always()
+        uses: actions/github-script@v5
+        env: 
+          TITLE: 'External Smoke Test Resutls:'
+          SMOKE_TEST_URL: ${{ secrets.SMOKE_TEST_URL }}
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const script = require( './package/woocommerce/packages/js/e2e-environment/bin/post-results-to-github-pr.js' )
```
Comment (rrennick) [inline .github/workflows/pr-smoke-test.yml:64]:
```suggestion
            const script = require( '@woocommerce/e2e-environment/bin/post-results-to-github-pr' )
```

I didn't test this change but it should work.

File: .github/workflows/pr-smoke-test.yml [hunk 4]
```diff
@@ -49,8 +51,50 @@ jobs:
...
+      - name: Post E2E tests results comment on PR
+        if: always()
+        uses: actions/github-script@v5
+        env: 
+          TITLE: 'External E2E Test Resutls:'
+          SMOKE_TEST_URL: ${{ secrets.SMOKE_TEST_URL }}
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const script = require( './package/woocommerce/packages/js/e2e-environment/bin/post-results-to-github-pr.js' )
```
Comment (rrennick) [inline .github/workflows/pr-smoke-test.yml:95]: Same suggestion here

**Review by rrennick — CHANGES_REQUESTED** [review]
Very nice @MrJnrman. I just had a couple small suggestions.

File: .github/workflows/pr-smoke-test.yml [hunk 5]
```diff
@@ -49,8 +51,50 @@ jobs:
...
+      - name: Post Smoke tests results comment on PR
+        if: always()
+        uses: actions/github-script@v5
+        env: 
+          TITLE: 'Smoke Test Results:'
+          SMOKE_TEST_URL: ${{ secrets.SMOKE_TEST_URL }}
+        with:
+          github-token: ${{secrets.GITHUB_TOKEN}}
+          script: |
+            const script = require( '@woocommerce/packages/js/e2e-environment/bin/post-results-to-github-pr' )
```
Comment (rrennick) [inline .github/workflows/pr-smoke-test.yml]: This still has the `packages/js/` in the package name/path so CI can''t find it. In this repo `@woocommerce/e2e-environment` maps to `packages/js/e2e-environment`.

Comment (MrJnrman) [reply .github/workflows/pr-smoke-test.yml]: Thanks for catching that 😅. The paths are now fixed.

File: .github/workflows/pr-smoke-test.yml (hunk 3 above)
Comment (rrennick) [reply .github/workflows/pr-smoke-test.yml:64]: @MrJnrman Sorry, I was wrong here. It looks like the script is running in the Github context vs the repo context. We'll need to restore your original code here.

**Comment by rrennick** [comment]
@MrJnrman Can you merge `trunk` to fix the smoke test error and resolve the .gitignore conflict?

**Comment by MrJnrman** [comment]
@rrennick This branch has been merged with `trunk`

**Review by rrennick — APPROVED** [review]
@MrJnrman Nice work.

**Pull request merged**